src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -72,6 +72,19 @@ private void RunWorker()
 
                 try
                 {
+#if TARGET_OSX || NATIVEAOT
+                    // On other platforms, when the underlying native thread is created,
+                    // the thread name is set to the name of the managed thread by another thread.
+                    // However, on OS X and NativeAOT (across all OSes), only the thread itself can set its name.
+                    // Therefore, by this point the native thread is still unnamed as it has not started yet.
+                    // See https://github.com/dotnet/runtime/issues/106464.
```

Comment:
```suggestion
```
The comment above is self-explanatory. I do not think there is any useful additional context provided in the issue.